patches/api/0449-Adding-PredicateChoice-to-Paper-API.patch

```diff
@@ -0,0 +1,187 @@
++                if (stack.getType() == match) {
++                    return true;
++                }
++            }
++            return false;
++        }
++    }
++
++    class ExactItemPredicate implements ItemPredicate {
++        private List<ItemStack> choices;
++
++        public ExactItemPredicate(@NotNull ItemStack stack) {
++            this(Arrays.asList(stack));
++        }
++
++        public ExactItemPredicate(@NotNull ItemStack... stacks) {
++            this(Arrays.asList(stacks));
++        }
++
++        public ExactItemPredicate(@NotNull List<ItemStack> choices) {
++            Preconditions.checkArgument(choices != null, "choices");
++            Preconditions.checkArgument(!choices.isEmpty(), "Must have at least one choice");
++            for (ItemStack choice : choices) {
++                Preconditions.checkArgument(choice != null, "Cannot have null choice");
++            }
++
++            this.choices = new ArrayList<>(choices);
++        }
++
++        @Override
++        public List<ItemStack> recipeBookExamples() {
++            return choices;
++        }
++
++        @Override
++        public boolean test(final ItemStack stack) {
++            for (ItemStack match : choices) {
++                if (stack.isSimilar(match)) {
++                    return true;
++                }
++            }
++            return false;
++        }
++    }
++}
+diff --git a/src/main/java/de/verdox/predicatechoice/PredicateChoice.java b/src/main/java/de/verdox/predicatechoice/PredicateChoice.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..6e83917aa79d9d5e3fd121dd961097a5a01d14ba
+--- /dev/null
++++ b/src/main/java/de/verdox/predicatechoice/PredicateChoice.java
+@@ -0,0 +1,50 @@
++package de.verdox.predicatechoice;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.RecipeChoice;
++import org.jetbrains.annotations.NotNull;
++import java.util.Collections;
++import java.util.List;
++import java.util.Objects;
++
++public record PredicateChoice(ItemPredicate itemPredicate) implements RecipeChoice {
++    public PredicateChoice {
++        Preconditions.checkArgument(itemPredicate != null, "itemPredicate");
++        Preconditions.checkArgument(!itemPredicate.recipeBookExamples().isEmpty(), "Must have at least one template");
++    }
++
++    @Override
++    public final boolean test(final ItemStack stack) {
++        return itemPredicate.test(stack);
++    }
++
++    @Override
++    public @NotNull ItemStack getItemStack() {
++        ItemStack stack = new ItemStack(itemPredicate.recipeBookExamples().get(0));
++        // For compat
++        if (itemPredicate.recipeBookExamples().size() > 1) {
++            stack.setDurability(Short.MAX_VALUE);
++        }
++
++        return stack;
++    }
++
++    @Override
++    public PredicateChoice clone() {
++        return new PredicateChoice(itemPredicate);
```

Comment:
Asking a non rhetorical question, does this violate the clone method contract? (reusing the predicate instance)